.gitignore

```diff
@@ -1,2 +1,6 @@
 c.out
-bin/*
\ No newline at end of file
+bin/*
+
+vendor/*
+go.mod
+go.sum
```

Comment:
why are these ignored?

Reply:
These were only ignored because the original repo didn't include them, I didnt know if they had a reason for not including them. 

I don't mind adding go.mod and go.sum to the repo.

Reply:
I investigated why the other library didn't have a go.mod/go.sum file, and it turns out it's because the module was created prior to go 1.14 and still is backward compatible to go 1.13, which doesn't have modules.

I added in these module files to follow best golang practices.